client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -494,9 +496,28 @@ export class CheckoutThankYou extends Component<
 		return { path: '/checkout/thank-you/no-site', properties: {} };
 	};
 
+	getMasterBar = () => {
+		const { translate } = this.props;
+		const purchases = getPurchases( this.props ).filter( ( purchase ) => ! isCredits( purchase ) );
+		const wasEcommercePlanPurchased = purchases.some( isEcommerce );
+
+		const siteId = this.props.selectedSite?.ID;
+		const siteSlug = this.props.selectedSite?.slug;
+
+		return (
+			<MasterbarStyled
+				onClick={ () => page( `/home/${ siteSlug ?? '' }` ) }
+				backText={ translate( 'Back to dashboard' ) }
+				canGoBack={ !! siteId && ! wasEcommercePlanPurchased }
```

Comment:
Unrelated to your PR, but while at it, could you include a comment to explain why we are hide the "Back" button for eCommerce plans? See: https://github.com/Automattic/wp-calypso/pull/88572#discussion_r1526774699 

```
// Back button is hidden for E-Commcerce Plans as a workaround to avoid taking users back to the loading page.
```